vumi/persist/riak_manager.py

```diff
@@ -122,7 +122,17 @@ def from_config(cls, config):
             mapreduce_timeout=mapreduce_timeout, store_versions=store_versions)
 
     def close_manager(self):
-        self.client.close()
+        if self._parent is None:
+            # Only top-level managers may close the client.
+            self.client.close()
```

Comment:
Logically, `close_manager()` means "I'm finished with this manager" rather than "clean up all the underlying resources". The thing that calls it doesn't necessarily know (or care) whether it has a top-level manager or not.

My initial implementation required sub-managers to be explicitly closed as well, but there are too many places (mostly in vumi-go) where we cache model objects or otherwise have sub-managers with indeterminate lifecycles.